### PR TITLE
Remove deprecation warnings

### PR DESCRIPTION
**Pull request by soc**

### CODE REVIEW

**Comment by soc** [comment]
```
 [mima]  * deprecated method
 [mima]    stringToTypeName(java.lang.String)scala.reflect.api.Names#TypeNameApi in
 [mima]    trait scala.reflect.api.Names does not have a correspondent in new version
 [mima]  * deprecated method
 [mima]    stringToTermName(java.lang.String)scala.reflect.api.Names#TermNameApi in
 [mima]    trait scala.reflect.api.Names does not have a correspondent in new version
```

Is `scala.reflect` still considered experimental, or do we need to keep those for 2.11.x?

**Comment by soc** [comment]
```
[ERROR] /home/jenkins/workspace/scala-2.11.x-integrate-ide/target/scala-refactoring/org.scala-refactoring.library/src/main/scala/scala/tools/refactoring/transformation/TreeFactory.scala:116: error: type mismatch;
[ERROR]  found   : String
[ERROR]  required: TreeFactory.this.global.TermName
[ERROR]     if (mods != NoMods) valOrVarDef setSymbol NoSymbol.newValue(name, newFlags = mods.flags) else valOrVarDef
[ERROR]                                                                 ^
[ERROR] one error found
```

I'll investigate. Maybe it makes sense to put the implicits back for now to not block @EECOLOR's work any further.

**Comment by soc** [comment]
https://github.com/scala-ide/scala-refactoring/pull/75

**Comment by soc** [comment]
@adriaanm Can you comment on whether scala.reflect still considered experimental, or do we need to keep those implicits for 2.11.x?

**Comment by soc** [comment]
Ok, I abandoned the removal of the implicits for 2.11.x, I'll bring that up again for 2.12.

**Comment by adriaanm** [comment]
yes, scala.reflect will remain experimental

**Comment by adriaanm** [comment]
Since the 2.11.6 deadline is extremely nigh, I've tentatively postponed this PR (along with other recent ones) to 2.11.7. I'll make a pass through reviewed 2.11.7 PRs on Friday and move them back to 2.11.6 where feasible.

In the mean time, feel free to let me know if you'd like some help getting this reviewed by then.

**Comment by soc** [comment]
```
[ERROR] /home/jenkins/workspace/scala-2.11.x-integrate-ide@2/target/scala-ide/org.scala-ide.sdt.spy/src/org/scalaide/spy/SpyView.scala:135: error: type mismatch;
[ERROR]  found   : scala.concurrent.Lock
[ERROR]  required: java.util.concurrent.locks.ReentrantLock
[ERROR]                 frame.createFrame(new scala.concurrent.Lock())
[ERROR]                                   ^
[ERROR] one error found
```

I guess I need to undo that too for now ...

File: src/compiler/scala/reflect/quasiquotes/Holes.scala [hunk 1]
```diff
@@ -132,7 +132,7 @@ trait Holes { self: Quasiquotes =>
     private def mapF(tree: Tree, f: Tree => Tree): Tree =
       if (f(Ident(TermName("x"))) equalsStructure Ident(TermName("x"))) tree
       else {
-        val x: TermName = c.freshName()
+        val x: TermName = TermName(c.freshName())
```
Comment (lrytz) [inline src/compiler/scala/reflect/quasiquotes/Holes.scala]: you could remove the type annotation, it was only there to drive implicit search

File: src/reflect/scala/reflect/api/Names.scala [hunk 2]
```diff
@@ -30,16 +30,16 @@ import scala.language.implicitConversions
  */
 trait Names {
   /** An implicit conversion from String to TermName.
-   *  Enables an alternative notation `"map": TermName` as opposed to `TermName("map")`.
-   *  @group Names
-   */
+  * Enables an alternative notation `"map": TermName` as opposed to `TermName("map")`.
```
Comment (lrytz) [inline src/reflect/scala/reflect/api/Names.scala]: This is not a deprecation change, and it makes the formatting inconsistent with the rest of the file

File: src/reflect/scala/reflect/internal/TreeGen.scala [hunk 3]
```diff
@@ -362,7 +362,7 @@ abstract class TreeGen {
         if (body forall treeInfo.isInterfaceMember) None
         else Some(
           atPos(wrappingPos(superPos, lvdefs)) (
-            DefDef(NoMods, nme.MIXIN_CONSTRUCTOR, Nil, ListOfNil, TypeTree(), Block(lvdefs, Literal(Constant())))))
+            DefDef(NoMods, nme.MIXIN_CONSTRUCTOR, Nil, ListOfNil, TypeTree(), Block(lvdefs, Literal(Constant(()))))))
```
Comment (lrytz) [inline src/reflect/scala/reflect/internal/TreeGen.scala:365]: what a joy! :)

**Comment by lrytz** [comment]
LGTM! thanks a lot for the patch. Would be nice if you could fix the points I mentioned, since it requires a rebase anyway.

File: src/reflect/scala/reflect/api/Names.scala (hunk 2 above)
Comment (soc) [inline src/reflect/scala/reflect/api/Names.scala]: Thanks, not sure what happened here ...

**Comment by soc** [comment]
@lrytz All done!

**Comment by soc** [comment]
Looks like some dependency issue:

> org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: Could not resolve project dependencies
> Caused by: org.apache.maven.artifact.resolver.ArtifactResolutionException: Unable to get dependency information for org.scala-lang:scala-library:jar:2.10.5-SNAPSHOT: Failed to retrieve POM for org.scala-lang:scala-library:jar:2.10.5-SNAPSHOT: Could not transfer artifact org.scala-lang:scala-library:pom:2.10.5-SNAPSHOT from/to typesafe-ide-Select a Scala profile (http://private-repo.typesafe.com/typesafe/ide-Select a Scala profile): IllegalArgumentException
>   org.scala-lang:scala-library:jar:2.10.5-SNAPSHOT

**Comment by soc** [comment]
Now it looks like everything compiled. :-)

**Pull request merged**